### ISSUE TITLE
Unit test failures on Linux 32 bit architectures

### ISSUE THREAD

**Issue by TC01** [milochristiansen/lua#7]
It seems that (on Fedora) the unit tests fail when ran on a 32 bit architecture (i.e. i686 and armv7l). Take a look [at this build](https://koji.fedoraproject.org/koji/taskinfo?taskID=18810051), specifically [this build log for i686](https://kojipkgs.fedoraproject.org//work/tasks/57/18810057/build.log):

```
+ go test -compiler gc -ldflags '' github.com/milochristiansen/lua
--- FAIL: TestAssign (0.18s)
	testhelp.go:74: Assertion Failed!
		  Stack Trace:
		    "(native code)"
		    "error": <line: 116>
FAIL
FAIL	github.com/milochristiansen/lua	0.249s
```

Is this expected/known? (I assume rubble is theoretically still supposed to work with 32-bit DF). If so, I can simply not build on 32-bit architectures.

**Comment by milochristiansen** [comment]
It *should* work fine, but obviously isn't...

I have no idea what could cause this, but once I get home I will surely be investigating! Hopefully the problem can be reproduced in Windows, but if not I can throw together a Linux environment...

Hopefully I'll have a fix on friday.

**Comment by milochristiansen** [comment]
OK, the issue was common to all 32 bit systems, and was an easy fix (Yay!).

The problem was me converting an int64 to an int without checking to see if it would fit first. AFAIK this was the only place in the entire VM where I used plain ints (and the only reason I did here is because I was indexing a slice).

This bug should not have affected Rubble in any way, as no Rubble addons used large integers as table keys.

Fixed in 1.1.3
